R/drop_duplicate_tips_random.R

```diff
@@ -0,0 +1,83 @@
+#' If a language tree has tips with duplicate tip labels, drop all but one at random. If there are tips which are dialects of the same language, you can choose to drop all but one.
+#'
+#' @param tree 	an object of class "phylo".
+#' @param merge_dialects a logical specifying whether to replace dialect tip labels with the glottocode of the language that is their parent, and then drop all but one
+#' @param random_seed integer specifying random seed
+#' @param LanguageTable data-frame of a cldf LanguageTable with columns for glottocodes per languoid and dialects language parent glottocode. If merge_dialects is set to TRUE, a LanguageTable is necessary. If NULL then Glottolog v4.8 is fetched from the internet and used.
+#' @param trim_tip_label_to_first_eight a logical specifying whether we should trim the tip label to the first 8 characters. If the tip labels contain more than just glottocodes, it is necessary to trim away what is not glottocodes. If set to TRUE, only the first 8 characters are retained. This is necessary for the EDGE-trees as they contain names as well (e.g. "bamu1257_Bamu_Kiwaian"). If the tip labels only consist of glottocodes, this parameter can be set to TRUE or FALSE without difference.
+#' @return tree without duplicates or multiple dialects for the same language.
+#' @export
+
+drop_duplicate_tips_random <- function(tree = NULL,
+                                      merge_dialects = TRUE,
+                                      random_seed = NULL,
+                                      LanguageTable = "Glottolog_4.8",
```

Comment:
I think there should be one shared way (function) to access the Glottolog LanguageTable, since this is also used in the language_level_* functionality. I also think this argument should have a more telling name, e.g. `GlottologLanguageTable`.

Reply:
Yeah, good point. Like with language_level_df, I only set it like this as a fallback but it's probably good to fetch specifically that sheet in some good way.

@SimonGreenhill decides.

Reply:
based on what @SimonGreenhill has said elsewhere, I'll set this to NULL.